pkg/routeagent/controllers/route/route.go

```diff
@@ -516,6 +520,11 @@ func (r *Controller) processNextPod() bool {
 			return fmt.Errorf("error retrieving submariner-route-agent pod object %s: %v", name, err)
 		}
 
+		if pod.Status.PodIP == "" {
+			r.podWorkqueue.Forget(obj)
+			return fmt.Errorf("submariner-route-agent pod %q does not have an ip-address yet", name)
```

Comment:
May be we should consider failing hard Fatalf if this happens more than X times so the route-agent is re-created ?

Reply:
IMHO this is not required as K8s would assign the IPAddress to the Pod in few ms/secs. This will be seen only when the route-agent pods are coming up.

Reply:
Ok, I guess that would be a broken k8s, may be it's too defensive.